### PR TITLE
Remove encryption at rest from unsupported serverless features

### PR DESCRIPTION
**Pull request by shannonbradshaw**
Fixes DOC-9777

### CODE REVIEW

**Comment by shannonbradshaw** [comment]
We might also need to update this page: https://www.cockroachlabs.com/docs/cockroachcloud/serverless-faqs#is-encryption-at-rest-enabled-on-cockroachdb-serverless (waiting on an answer from eng).

**Comment by shannonbradshaw** [comment]
I don't think we need to wait further on eng. Can merge once this is approved.

**Comment by mdlinville** [comment]
PTAL at the commit I just pushed as as proposal for keeping the info in the page but being more clear. I also found a few other improvements to make nearby (we missed adding some unsupported features around PCI DSS / HIPAA, and we were using an old variant of a Liquid variable).

File: src/current/cockroachcloud/serverless-unsupported-features.md [hunk 1]
```diff
@@ -62,10 +62,8 @@ There is no self-service way of accessing [audit logs]({% link cockroachcloud/sq
 
 ## Encryption
 
-[Encryption at rest](https://www.cockroachlabs.com/docs/{{site.current_cloud_version}}/security-reference/encryption#encryption-at-rest) is not supported in CockroachDB {{ site.data.products.serverless }} clusters.
-
-[Customer-managed encryption keys]({% link cockroachcloud/managing-cmek.md %}) (CMEK) are not supported in CockroachDB {{ site.data.products.serverless }} clusters.
+Cluster storage for CockroachDB {{ site.data.products.serverless }} is encrypted at rest by the Cloud provider. [Customer-managed encryption keys]({% link cockroachcloud/managing-cmek.md %}) (CMEK) are not supported in CockroachDB {{ site.data.products.serverless }}. CMEK is available only in [CockroachDB {{ site.data.products.dedicated }} advanced]({% link cockroachcloud/create-your-cluster.md %}).
```
Comment (mdlinville) [inline src/current/cockroachcloud/serverless-unsupported-features.md:65]:
```suggestion
Cluster storage for CockroachDB {{ site.data.products.serverless }} is encrypted at rest by the cloud provider. [Customer-managed encryption keys]({% link cockroachcloud/managing-cmek.md %}) (CMEK) are not supported in CockroachDB {{ site.data.products.serverless }}. CMEK is available only in [CockroachDB {{ site.data.products.dedicated }} advanced]({% link cockroachcloud/create-your-cluster.md %}).
```

**Review by mdlinville — APPROVED** [review]
LGTM either way

**Pull request merged**